lib/open_project/translations/models/locales_updater.rb

```diff
@@ -0,0 +1,178 @@
+# todo those ought to be in the gemspec
+require 'tempfile'
+require 'crowdin-api'
+require 'yaml'
+require 'zip'
+require 'pathname'
+require 'fileutils'
+
+require_relative '../helpers/tmp_directory'
+require_relative './git_repository'
+
+class LocalesUpdater
+  extend TmpDirectory
+
+  def self.update_all_locales_of_all_repos(debug: true)
+    repos_to_update = plugins_with_locales
+
+    within_tmp_directory(delete_if_exists: true, debug: debug) do
+      repos_to_update.each do |name, specifics|
+        # todo each branch
+        uri = specifics[:uri]
+        set_crowdin_specifics(specifics)
+
+        within_tmp_directory(path: File.join(FileUtils.pwd, name), debug: debug) do
+          git_repo = GitRepository.new(uri, FileUtils.pwd)
+          git_repo.clone_or_pull
+
+          git_repo.checkout(branch)
+          # todo or should we merge this branch into the next ('push vs pull')
+          git_repo.merge(previous_branch, '-Xours') if previous_branch
+          git_repo.within_repo do
+            begin
+              upload_english
+              request_build
+              download_and_replace_locales
+            rescue Crowdin::API::Errors::Error => e
+              puts "Error during update of #{name}: #{e.message}"
+            end
+            fix_missing_keys
+          end
+          git_repo.add('config/locales')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.